### PR TITLE
Fix: Renovate/major all dependencies

### PR DESCRIPTION
**Pull request by jamsinclair**
Attempting to fix the `renovate/major-all-dependencies` PR as noted in #152.

## Changes
- Fix package `lint` script. Removes the missing directory `features`
- Fix package `test-unit` script. Update unit tests to use `enzyme-adapter`
- Downgrade to `html-reporter` v4 as later versions do not support gemini 
- Downgrade `babel-loader`. The migration to `@babel v7` would be better handled in separate task. 
- Fix building with Webpack v4
	- Migrate from `JSUglifyPlugin` to `TerserPlugin`
	- Migrate from `ExtractTextPlugin` to `MiniCssExtractPlugin`

## Notes

The only affected change I've noticed is Webpack is collapsing the css styles to one liners in `lib/style.css`. Not sure if Webpack itself or plugin, only occurs in `production` mode.

Also was the `example` script removed? I was unable to get regression tests running.

### CODE REVIEW

**Comment by jamsinclair** [comment]
Looks like CircleCI is running on non-master fork PRs. 

Your CI is passing over at: https://circleci.com/gh/jamsinclair/react-carousel/3

File: package.json [hunk 1]
```diff
@@ -46,28 +45,29 @@
...
+    "enzyme-adapter-react-16": "^1.15.2",
     "eslint": "^6.8.0",
     "eslint-config-brainhub": "^1.11.1",
     "eslint-watch": "^6.0.1",
     "express": "^4.17.1",
-    "extract-text-webpack-plugin": "^3.0.2",
     "file-loader": "^5.0.2",
     "gemini": "^7.5.2",
-    "html-reporter": "^5.8.0",
+    "html-reporter": "4.13.2",
```
Comment (piotr-s-brainhub) [inline package.json]: why do you downgrade this?

File: package.json [hunk 2]
```diff
@@ -25,13 +25,12 @@
...
   "author": "Brainhub",
   "license": "MIT",
-  "dependencies": {},
   "devDependencies": {
     "autoprefixer": "^9.7.3",
     "babel-cli": "^6.26.0",
     "babel-core": "^6.26.3",
     "babel-eslint": "^10.0.3",
-    "babel-loader": "^8.0.6",
+    "babel-loader": "^7.1.5",
```
Comment (piotr-s-brainhub) [inline package.json]: why do you downgrade this?

File: package.json (hunk 1 above)
Comment (jamsinclair) [reply package.json]: Downgraded html-reporter to v4 as later versions do not support the gemini package (used by your `test-regression` script).

File: package.json (hunk 2 above)
Comment (jamsinclair) [reply package.json]: Downgraded babel-loader because v8 only supports Babel 7. 

I think migrating to Babel 7 would be better handled as a separate task.

**Comment by piotr-s-brainhub** [comment]
could you resolve the conflicts, please?

**Comment by jamsinclair** [comment]
Closing PR to make a new one with a better diff. The rebasing nature of your Renovate bot has messed it up. Easier to start a fresh.

**Pull request closed without merging**